tests/unit/test_decide_user_install.py

```diff
@@ -0,0 +1,124 @@
+"""Test user site-packages installation decision
+and other install destination option conflicts.
+"""
+
+from itertools import product
+
+from pytest import mark, raises
+
+from pip._internal.commands.install import decide_user_install
+from pip._internal.exceptions import CommandError, InstallationError
+
+ENABLE_USER_SITE = 'pip._internal.commands.install.ENABLE_USER_SITE'
+ISDIR = 'pip._internal.commands.install.os.path.isdir'
+EXISTS = 'pip._internal.commands.install.os.path.exists'
+WRITABLE = 'pip._internal.commands.install.test_writable_dir'
+VIRTUALENV_NO_GLOBAL = 'pip._internal.commands.install.virtualenv_no_global'
+
+
+def false(*args, **kwargs):
+    """Return False."""
+    return False
+
+
+def true(*args, **kwargs):
+    """Return True."""
+    return True
+
+
+@mark.parametrize(('use_user_site', 'prefix_path', 'target_dir', 'root_path'),
+                  filter(lambda args: sum(map(bool, args)) > 1,
+                         product((False, True), (None, 'foo'),
+                                 (None, 'bar'), (None, 'baz'))))
+def test_conflicts(use_user_site, prefix_path, target_dir, root_path):
+    """Test conflicts of target, user, root and prefix options."""
+    with raises(CommandError):
+        decide_user_install(
+            use_user_site=use_user_site, prefix_path=prefix_path,
+            target_dir=target_dir, root_path=root_path)
+
+
+def test_target_exists(monkeypatch):
+    """Test exist target path."""
+    monkeypatch.setattr(WRITABLE, true)
```

Comment:
Looks like you repeat this a lot. How about moving it into a separate fixture?

Reply:
Thank you, I did not know that it was possible to do that (or TBH I was carelessly readding pytest docs :grin:).

Reply:
Yeah, you can also cascade fixtures so if the combination of simple fixtures is used often, you could combine them into another fixture and use that...